schema-registry/rest-service/src/main/java/com/hortonworks/registries/schemaregistry/webservice/ConfluentSchemaRegistryCompatibleResource.java

```diff
@@ -201,18 +201,42 @@ public Response getSchemaVersion(@ApiParam(value = "subject", required = true)
                                              String versionId) {
         Response response;
         try {
+            SchemaVersionInfo schemaVersionInfo = null;
 
-            SchemaVersionInfo schemaVersionInfo = "latest".equals(versionId)
-                    ? schemaRegistry.getLatestSchemaVersionInfo(subject)
-                    : schemaRegistry.getSchemaVersionInfo(Long.valueOf(versionId));
+            if ("latest".equals(versionId)) {
```

Comment:
we should probably add some versionId validation and throw invalidVersionException() if not valid input range.
see here for logic of the id validation https://github.com/confluentinc/schema-registry/blob/8eb664dbc84b1c2db3666fa0771eeb0e0909f892/core/src/main/java/io/confluent/kafka/schemaregistry/rest/VersionId.java